tests/test_util/aiohttp.py

```diff
@@ -270,3 +281,50 @@ def create_session(hass, *args):
         side_effect=create_session,
     ):
         yield mocker
+
+
+class MockLongPollSideEffect:
+    """Imitate a long_poll request. Once created, actual responses are queued and if queue is empty, will await until done."""
+
+    def __init__(self):
+        """Initialize the queue."""
+        self.semaphore = asyncio.Semaphore(0)
+        self.response_list = []
+        self.stopping = False
+
+    async def __call__(self, method, url, data):
+        """Fetch the next response from the queue or wait until the queue has items."""
+        if self.stopping:
+            raise ClientError()
+        await self.semaphore.acquire()
+        (status, content, cookies, exc, headers) = self.response_list.pop(0)
+        return AiohttpClientMockResponse(
+            method, url, status, content, cookies, exc, headers
```

Comment:
Maybe push a dict into the list and then use it as kwargs?

Return aiocmockresponse(method, url, **thatdict)

Reply:
i like the idea. let me try

Reply:
ok. done